### PR TITLE
Change/migrate micro deps to java7

### PR DESCRIPTION
**Pull request by pilloPl**
migration of a few classes from micro deps module to java

### CODE REVIEW

File: micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/watcher/DependencyState.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.ofg.infrastructure.discovery.watcher;
+
+enum DependencyState {
```
Comment (marcingrzejszczak) [inline micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/watcher/DependencyState.java]: it's package scope deliberately?

File: micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/watcher/DependencyWatcher.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    public void registerDependencyStateChangeListener(DependencyWatcherListener listener) {
+        listeners.add(listener);
+    }
+
+    public void unregisterDependencyStateChangeListener(DependencyWatcherListener listener) {
+        listeners.remove(listener);
+    }
+
+    private final List<MicroserviceConfiguration.Dependency> dependencies;
```
Comment (marcingrzejszczak) [inline micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/watcher/DependencyWatcher.java]: can you move that to the top?

**Comment by marcingrzejszczak** [comment]
Apart from the comments +1

File: micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/watcher/DependencyWatcher.java (hunk 2 above)
Comment (pilloPl) [inline micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/watcher/DependencyWatcher.java]: Yep, sure. It is stupid IDEA refractoring

File: micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/watcher/DependencyState.java (hunk 1 above)
Comment (pilloPl) [inline micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/watcher/DependencyState.java]: Yes. Why not? I know it was public, but as far as I remember It was only used in this package (despite tests, but spock handles that)

Comment (marcingrzejszczak) [inline micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/watcher/DependencyState.java]: Just asking - I saw that in Groovy it was public. I don't remember the case where people will try to create their own dependency watchers. Will they not need this enum? Regardless of where they watcher lays?

Comment (pilloPl) [inline micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/watcher/DependencyState.java]: You might be right - my way of thinking was like following:
only this package uses that so I changed to package scope

Comment (marcingrzejszczak) [inline micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/watcher/DependencyState.java]: So please make it public and merge this :)

Comment (pilloPl) [inline micro-deps/micro-deps/src/main/groovy/com/ofg/infrastructure/discovery/watcher/DependencyState.java]: Changed to public scope.

**Comment by marcingrzejszczak** [comment]
Please check if your changes haven't broken the acceptance tests - https://travis-ci.org/4finance/micro-infra-spring

**Pull request merged**